samples/connectedhomeip/light_bulb/src/LightBulbPublishService.cpp

```diff
@@ -0,0 +1,145 @@
+// PublishTimerHandler method
+static LightBulbPublishService* sLightBulbPublishService;
+
+const char PUBLISH_SERVICE_PAYLOAD[] = "chip-light-bulb";
+const char PUBLISH_DEST_ADDR[] = "ff02::1";
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::Transport;
+using namespace ::chip::DeviceLayer;
+
+void LightBulbPublishService::Init()
+{
+    // Set pointer to the current instance
+    sLightBulbPublishService = this;
+    mIsServiceStarted = false;
+
+    // Initialize timer
+    k_timer_init(&mPublishTimer, &PublishTimerHandler, nullptr);
+    k_timer_user_data_set(&mPublishTimer, this);
+
+    // Initialize UDP endpoint
+    UdpListenParameters udpParams(&chip::DeviceLayer::InetLayer);
+    udpParams.SetAddressType(kIPAddressType_IPv6);
+    udpParams.SetInterfaceId(INET_NULL_INTERFACEID);
+    udpParams.SetListenPort(SERVICE_UDP_PORT);
+    if (mUdp.Init(udpParams))
+    {
+        LOG_ERR("LightBulbPublishService initialization failed");
+    }
+
+    mMessageId = 0;
+    IPAddress::FromString(PUBLISH_DEST_ADDR, mPublishDestAddr);
+}
+
+void LightBulbPublishService::Start(uint32_t aTimeoutInMs, uint32_t aIntervalInMs)
+{
+    if (!mIsServiceStarted)
+    {
+        if (aTimeoutInMs >= aIntervalInMs)
+        {
+            LOG_INF("Started Publish service");
+
+            mPublishInterval = aIntervalInMs;
+            // Calculate number of publish iterations to do considering given timeout and interval values.
+            mPublishIterations = aTimeoutInMs / aIntervalInMs;
+            mIsServiceStarted = true;
+
+            k_timer_start(&mPublishTimer, K_MSEC(aIntervalInMs), K_NO_WAIT);
+        }
+        else
+        {
+            LOG_ERR("Publish service cannot be started: service timeout is greater than interval");
+        }
+    }
+    else
+    {
+        LOG_INF("Publish service is already running");
+    }
+}
+
+void LightBulbPublishService::Stop()
+{
+    if (mIsServiceStarted)
+    {
+        LOG_INF("Stopped Publish service");
+
+        k_timer_stop(&mPublishTimer);
+
+        mIsServiceStarted = false;
+        mMessageId = 0;
+    }
+    else
+    {
+        LOG_INF("Publish service is not running");
+    }
+}
+
+void LightBulbPublishService::Publish()
+{
+    if (!chip::DeviceLayer::ConnectivityMgrImpl().IsThreadAttached())
+    {
+        return;
+    }
+
+    const uint16_t payloadLength = sizeof(PUBLISH_SERVICE_PAYLOAD);
+
+    chip::System::PacketBufferHandle buffer = chip::System::PacketBuffer::NewWithAvailableSize(payloadLength);
+    if (!buffer.IsNull())
+    {
+        memcpy(buffer->Start(), PUBLISH_SERVICE_PAYLOAD, payloadLength);
+        buffer->SetDataLength(payloadLength);
+
+        // Subsequent messages are assumed to have increasing IDs.
+        PacketHeader header;
+        header.SetMessageId(mMessageId++);
+
+        if (!(mUdp.SendMessage(header, Transport::PeerAddress::UDP(mPublishDestAddr, SERVICE_UDP_PORT), std::move(buffer))))
```

Comment:
`if (SendMessage() == CHIP_NO_ERROR)` would be more informative, i.e. would let a reader know that the return type of the function is CHIP_ERROR.